### PR TITLE
Include how to run static checks in dev docs

### PR DESCRIPTION
**Pull request by cpeel**
Add phpstan to our ci Makefile and give instructions on how to run the static checks as part of our dev docs. I needed to update the phpstan temp directory such that it worked on TEST which is a shared dev environment.

FYI @mrducky4 & @jchaffraix

### CODE REVIEW

**Comment by srjfoo** [comment]
I think [this](https://github.com/DistributedProofreaders/dproofreaders/pull/1272/files#diff-e66e6614eeae65eceeb948d27352017fa514337dbcce84ccfa6f851c049d277cL65) should probably be "Principles".

I often have to check usage for this pair:
> If you find yourself having trouble choosing between principal and principle, think about the context. Use principal in reference to a person who is in leadership or to describe the importance of something; use principle to refer to a standard, rule, or guiding belief.

**Review by srjfoo — APPROVED** [review]
Most of this is way above my head, so the only thing I could really help with was the documentation.

**Comment by cpeel** [comment]
> I think [this](https://github.com/DistributedProofreaders/dproofreaders/pull/1272/files#diff-e66e6614eeae65eceeb948d27352017fa514337dbcce84ccfa6f851c049d277cL65) should probably be "Principles".

It is, thanks. I get this and effect/affect wrong all the time.

**Review by srjfoo — APPROVED** [review]

**Review by chrismiceli — APPROVED** [review]

**Review by jchaffraix — APPROVED** [review]
2 questions/comments that can be addressed as follow-ups and may just be me not knowing the context of how we run our CI.

File: SETUP/ci/Makefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
-.PHONY: all less security_checks best_practice_checks lint_charsuites lint_css lint_code lint tests
+.PHONY: all less security_checks best_practice_checks static_checks lint_charsuites lint_css lint_code lint less tests shellcheck
 
 SELF_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
 
-all: less security_checks best_practice_checks lint
+all: security_checks best_practice_checks lint static_checks
```
[the remark this answers is not in the document]
Comment (cpeel) [reply SETUP/ci/Makefile:5]: It is. `shellcheck` is an external program and not a composer package or one of our primary dependencies (eg PHP) that the other ones use. It's also pretty rare that we make changes to the shell scripts and the CI will run in as part of a PR anyway.

File: SETUP/ci/check_require_login.php [hunk 2]
```diff
@@ -81,6 +81,11 @@
         continue;
     }
 
+    // If it's in the phpstan cache directory, skip it
+    if (startswith($file, ".phpstan.cache/")) {
+        continue;
+    }
+
```
[the remark this answers is not in the document]
Comment (cpeel) [reply SETUP/ci/check_require_login.php:88]: No, we want to check all files that will be accessible on the web. We [manually exclude](https://github.com/DistributedProofreaders/dproofreaders/blob/master/SETUP/ci/check_require_login.php#L41) most of the applicable files from `pinc/3rdparty` that we know to be OK, but this forces us to inspect and confirm any new files we add to that path.

**Review by 70ray — APPROVED** [review]

**Pull request merged**